code/src/reducers/game.js

```diff
@@ -0,0 +1,69 @@
+import { createSlice } from '@reduxjs/toolkit';
+import ui from './ui';
+
+export const game = createSlice({
+    name: 'game',
+    initialState: {
+        username: null,
+        currentLocation: null,
+        type: '',
+        direction: '',
+    },
+
+    reducers: {
+        setUsername: (store, action) => {
+            store.username = action.payload;
+        },
+        setCurrentLocation: (store, action) => {
+            store.currentLocation = action.payload;
+        },
+        setType: (store, action) => {
+            store.type = action.payload;
+        },
+        setDirection: (store, action) => {
+            store.direction = action.payload;
+        }
+
+    }
+});
+
+export const generateMaze = () => {
+    return (dispatch, getState) => {
+        dispatch(ui.actions.setLoading(true));
+
+        const settings = {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ username: getState().game.username }),
+        };
+        fetch("https://labyrinth-technigo.herokuapp.com/start", settings)
+            .then((res) => res.json())
+            .then((json) => {
+                setTimeout(() => dispatch(game.actions.setCurrentLocation(json)))
+                setTimeout(() => dispatch(ui.actions.setLoading(false)), 1000)
+            })
```

Comment:
this time delay was super!! We didn't figure this out so will add if we have time too since it looks fantastic on your loading! : D